cast/js/src/main/resources/preamble.js

```diff
@@ -646,4 +706,55 @@ function ActiveXObject() {
 	}
 }
 
+CSSStyleDeclaration = function CSSStyleDeclaration() {
+	this.getPropertyValue = function CSSStyleDeclaration_getPropertyValue(property) {
+		// TODO: model me
+	}
+};
+
+ArrayBuffer = function ArrayBuffer() {};
 
+Uint8Array = function Uint8Array() {};
+Uint8Array.prototype.subarray = function Uint8Array_prototype_subarray(begin,end) {
+	// TODO: model me
+};
+
+Generator = function Generator() {};
+Generator.prototype.next = function Generator_prototype_next() {
+	// TODO: model me
+};
+
+History = function History() {};
+History.prototype.pushState = function History_prototype_pushState() {
+	// TODO: model me
+};
+History.prototype.replaceState = function History_prototype_replaceState() {
+	// TODO: model me
+};
+
+LocalStorage = function LocalStorage(){
+	this.setItem = function localStorage_setItem(keyName, keyValue){
+		// TODO: model me
+	},
+	this.getItem = function localStorage_getItem(keyName){
+		// TODO: model me
+	},
+	this.removeItem = function localStorage_removeItem(keyName){
+		// TODO: model me
+	},
+	this.clear = function localStorage_clear(){
+		// TODO: model me
+	}
+}
+
+Console = function Console(){
```

Comment:
Does this help with calls like `console.log(...)`?  If so, how, given that the variable is named `Console`?

Reply:
Yes. Sample Program:
```

function getRandomNumber(min, max) {
        return Math.floor(Math.random() * (max - min + 1)) + min;
}

var randomNumber = getRandomNumber(1, 100);

console.log("Random Number:", randomNumber);

if (randomNumber % 2 === 0) {
        console.debug("The random number is even.");
} else {
        console.info("The random number is odd.");
}
    
```
 Output: 
 
```
   "test.js@1:0-341": {
    "test.js@12:297-338": [
      "console_info (Native)"
    ],
    "test.js@5:125-148": [
      "test.js@1:0-104"
    ],
    "test.js@7:151-194": [
      "console_log (Native)",
      "Math_log (Native)"
    ],
    "test.js@10:235-278": [
      "console_debug (Native)"
    ]
  }
```

Reply:
Thanks, but can you explain why adding a write to the `Console` variable leads to new information for a variable named `console`?  These are two different variables.